files/bsd/rc.local.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/sh
+
+# Save to /etc/rc.local, to be ran on startup
+
+if [ "$(uname)" = "NetBSD" ]
+then
+    export PATH=/usr/bin:/bin:/usr/sbin:/sbin:/usr/pkg/bin:/usr/pkg/sbin
+
+    # restarting syslogd
+    restart_syslogd()
+    {
+        /etc/rc.d/syslogd restart > /dev/null
+    }
+    
+    # chown home paths
+    chown_home()
+    {
+        chown -R ${username}:users /home/${username}
+    }
+elif [ "$(uname)" = "OpenBSD" ]
+then
+    export PATH=/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/local/sbin
+
+    # restarting syslogd
+    restart_syslogd()
+    {
+        rcctl restart syslogd > /dev/null
+    }
+
+    # chown home paths
+    chown_home()
+    {
+        chown -R ${username}:${username} /home/${username}
+    }
+else
+    # Unsupported OS
+    echo "Unsupported OS, exiting"
+    exit 1
+fi
+
+### get instance hostname ###
+
+instance_hostname=$(curl -s -H "Metadata-Flavor: Google"  http://metadata.google.internal/computeMetadata/v1/instance/name)
+current_hostname=$(hostname)
+
+if [ "$instance_hostname" != "" ]
+then
+    if [ "$instance_hostname" != "$current_hostname" ]
+    then
+        echo "Setting hostname to $instance_hostname"
+        hostname $instance_hostname
+        echo $instance_hostname > /etc/myname
+        restart_syslogd
+    else
+        echo "Hostname is correct"
+    fi
+else
+    echo "Could not discover hostname"
+fi
+
+### set up user's keys ###
+
+instance_keys=$(curl -s -H "Metadata-Flavor: Google"  http://metadata.google.internal/computeMetadata/v1/project/attributes/ssh-keys)
+packer_temp_root_key=$(curl -s -H "Metadata-Flavor: Google"  http://metadata.google.internal/computeMetadata/v1/instance/attributes/ssh-keys)
```

Comment:
Hm - it doesn't seem quite right to reference packer here, it's not packer specific. And the instance_keys name, in the previous line, seems a bit misleading too, given that it's fetch project level keys? Perhaps worth a a comment or two about fetching project and instance level keys?